### PR TITLE
[5.1] HasOneThrough

### PR DESCRIPTION
**Pull request by JapSeyz**
Credit to https://github.com/pandaac/apolune/blob/09785a40e1b9070176b75aaf2a982ab20297697d/src/Apolune/Core/Database/Eloquent/Relations/HasOneThrough.php

### CODE REVIEW

**Comment by Sirlazer2** [comment]
Sorry for the many minor fix commits.

**Comment by taylorotwell** [comment]
What table structure is this attempting to solve?

File: src/Illuminate/Database/Eloquent/Model.php [hunk 1]
```diff
@@ -774,6 +775,26 @@ public function hasOne($related, $foreignKey = null, $localKey = null)
         return new HasOne($instance->newQuery(), $this, $instance->getTable().'.'.$foreignKey, $localKey);
     }
 
+    /*
+    * Define a has-one-through relationship.
+    *
+    * @param  string  $related
+    * @param  string  $through
+    * @param  string|null  $firstKey
+    * @param  string|null  $secondKey
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: missing local key

File: src/Illuminate/Database/Eloquent/Model.php [hunk 2]
```diff
@@ -774,6 +775,26 @@ public function hasOne($related, $foreignKey = null, $localKey = null)
...
+    * @return \Illuminate\Database\Eloquent\Relations\HasOneThrough
+    */
+
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: no

File: src/Illuminate/Database/Eloquent/Model.php [hunk 3]
```diff
@@ -774,6 +775,26 @@ public function hasOne($related, $foreignKey = null, $localKey = null)
         return new HasOne($instance->newQuery(), $this, $instance->getTable().'.'.$foreignKey, $localKey);
     }
 
+    /*
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: incorrect

File: src/Illuminate/Database/Eloquent/Model.php [hunk 4]
```diff
@@ -774,6 +775,26 @@ public function hasOne($related, $foreignKey = null, $localKey = null)
...
+    * @param  string  $related
+    * @param  string  $through
+    * @param  string|null  $firstKey
+    * @param  string|null  $secondKey
+    * @return \Illuminate\Database\Eloquent\Relations\HasOneThrough
+    */
+
+    public function hasOneThrough($related, $through, $firstKey = null, $secondKey = null, $localKey = null)
+    {
+        $through = app($through);
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Eloquent/Model.php]: no way you can use the app function here!

**Comment by Sirlazer2** [comment]
*users*
id

*shops*
id
user_id
chain_id
shop_identifier

*chains*
id
shop_name


This allows the user to do $user->chain as the user can only moderate one shop-chain, but multiple shops.

**Comment by Sirlazer2** [comment]
But shop is not a pivot table per say, because there are shop managers and chain managers. 

The shop manager will not have access to the chain, therfore it cannot be used sloely as a pivot table

**Comment by GrahamCampbell** [comment]
> Credit to

You should author the first commit as that user, then apply your changes on top.

**Comment by Sirlazer2** [comment]
Sorry, I am a bit new to the whole pull request etc. 

And didn't know you could that with styleci. 

I think I may have caused more problems they it's worth, therefore I could probably just turn it into a package and use it myself.

**Pull request closed without merging**